Slim/Plugin/AudioScrobbler/Plugin.pm

```diff
@@ -812,22 +1038,46 @@ sub submitScrobble {
 
 		return $cb->();
 	}
-
-	if ( !$client->master->pluginData('submit_url') ) {
-		# Get a new session
-		handshake( {
-			client => $client,
-			cb     => sub {
-				submitScrobble( $client, $params );
-			},
-		} );
-		return;
-	}
-
-	if ( main::DEBUGLOG && $log->is_debug ) {
-		$log->debug( 'Scrobbling ' . scalar( @{$queue} ) . ' queued item(s)' );
-		#$log->debug( Data::Dump::dump($queue) );
-	}
+       # Find out the API type and call ListenBrainz if necessary
+       my $accounts = getAccounts($client);
+       my ($api_type, $api_url, $api_key);
+       for my $acc (@{$accounts}) {
+	       if ($acc->{username} eq $account) {
+		       $api_type = $acc->{api_type} || 'lastfm';
+		       $api_url  = $acc->{api_url}  || '';
+		       $api_key  = $acc->{password} || '';
+		       last;
+	       }
+       }
+       if ( main::DEBUGLOG && $log->is_debug ) {
```

Comment:
Please make sure you always use tabs rather than spaces for indentation. Everywhere.